appinventor/appengine/src/com/google/appinventor/client/explorer/folder/FolderManager.java

```diff
@@ -0,0 +1,193 @@
+    LOG.info("Created new folder manager");
+  }
+
+  public void loadFolders() {
+    String foldersAsString = Ode.getUserSettings()
+        .getSettings(SettingsConstants.USER_GENERAL_SETTINGS)
+        .getPropertyValue(SettingsConstants.FOLDERS);
+    foldersLoaded = true;
+
+    if (foldersAsString.isEmpty()) {
+      LOG.info("Initialize folders");
+      initializeFolders();
+      fireFoldersLoaded();
+      return;
+    }
+
+    JSONObject folderJSON = JSONParser.parse(foldersAsString).isObject();
+    if (folderJSON.get(FolderJSONKeys.PROJECTS).isArray().size() == 0 &&
+        folderJSON.get(FolderJSONKeys.CHILD_FOLDERS).isArray().size() == 0) {
+      LOG.info("Global folder is empty");
+      initializeFolders();
+      fireFoldersLoaded();
+      return;
+    }
+
+    LOG.info("folderJSON - " + folderJSON);
+    globalFolder = new ProjectFolder(folderJSON, null);
+    LOG.info("Creating Trash Folder");
+    trashFolder = globalFolder.getChildFolder(FolderJSONKeys.TRASH_FOLDER);
+    checkForUnassignedProjects();
+    fireFoldersLoaded();
+  }
+
+  public void saveAllFolders() {
+    LOG.info("Saved Folder JSON: " + globalFolder.toJSON().toString());
+
+    Ode.getUserSettings()
+        .getSettings(SettingsConstants.USER_GENERAL_SETTINGS)
+        .changePropertyValue(SettingsConstants.FOLDERS, globalFolder.toJSON().toString());
+    Ode.getUserSettings().saveSettings(null);
+  }
+
+  public ProjectFolder createFolder(String name, ProjectFolder parent) {
+    ProjectFolder folder = new ProjectFolder(name, System.currentTimeMillis(), System.currentTimeMillis(), parent);
+    parent.addChildFolder(folder);
+    while ((parent = parent.getParentFolder()) != null) {
+      parent.clearCache();
+    }
+    saveAllFolders();
+    fireFolderAdded(folder);
+    return folder;
+  }
+
+  public void renameFolders(List<String> folderNames, List<ProjectFolder> foldersToRename) {
+    for (int i = 0; i < foldersToRename.size(); ++i) {
+      foldersToRename.get(i).setName(folderNames.get(i));
+    }
+    saveAllFolders();
+  }
+
+  public void moveItemsToFolder(List<Project> projects, List<ProjectFolder> folders, ProjectFolder destination) {
+    LOG.info("Moving projects count " + projects.size() + " to " + destination.getName());
+    for (Project project : projects) {
+      LOG.info("Moving project " + project.getProjectName()  + " from " + project.getHomeFolder().getName() +
+                   " to " + destination.getName());
+      destination.addProject(project);
+    }
+    for (ProjectFolder folder : folders) {
+      LOG.info("Moving folder " + folder.getName()  + " from " + folder.getParentFolder().getName() +
+                   " to " + destination.getName());
+      destination.addChildFolder(folder);
+    }
+    saveAllFolders();
+    fireFoldersChanged();
+  }
+
+  // relative to *global*
+  public ProjectFolder createFolder(String path) {
+    return null;
```

Comment:
This method isn't used, consider removing it (and what does it mean to return null when creating a folder?).